frontend/src/lib/components/Forms/FolderTreeSelect.svelte

```
@@ -0,0 +1,383 @@
+	//  Flat search results (DFS)
+	type SearchResult = { node: TreeNode; path: string[] };
+
+	const searchData = $derived.by(() => {
+		const q = debouncedQuery.trim().toLowerCase();
+		if (!q) return null;
+		const results: SearchResult[] = [];
+		function visit(n: TreeNode, ancestors: string[]) {
+			if (n.uuid !== excludedId && n.uuid && n.name.toLowerCase().includes(q)) {
+				results.push({ node: n, path: ancestors });
+			}
+			(n.children ?? []).forEach((c) => visit(c, [...ancestors, n.name]));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)